capsules/src/net/udp/driver.rs

```diff
@@ -466,154 +436,176 @@ impl<'a> LegacyDriver for UDPDriver<'a> {
-                        .copy_from_slice(&self.interface_list[i].0);
-                }
-                // Returns total number of interfaces
-                ReturnCode::SuccessWithValue {
-                    value: self.interface_list.len(),
+            1 => {
+                let res = self
+                    .apps
+                    .enter(appid, |app, _| {
+                        app.app_cfg.mut_map_or(Err(ErrorCode::INVAL), |cfg| {
+                            if cfg.len() != arg1 * mem::size_of::<IPAddr>() {
+                                return Err(ErrorCode::INVAL);
+                            }
+                            let n_ifaces_to_copy = cmp::min(arg1, self.interface_list.len());
+                            let iface_size = mem::size_of::<IPAddr>();
+                            for i in 0..n_ifaces_to_copy {
+                                cfg[i * iface_size..(i + 1) * iface_size]
+                                    .copy_from_slice(&self.interface_list[i].0);
+                            }
+                            // Returns total number of interfaces
+                            Ok(self.interface_list.len())
+                        })
+                    })
+                    .unwrap_or_else(|err| Err(err.into()));
+                match res {
+                    Ok(v) => CommandResult::success_u32(v as u32),
+                    Err(e) => CommandResult::failure(e),
                 }
-            }),
+            }
 
             // Transmits UDP packet stored in tx_buf
             2 => {
-                self.do_with_app(appid, |app| {
-                    if app.pending_tx.is_some() {
-                        // Cannot support more than one pending tx per process.
-                        return ReturnCode::EBUSY;
-                    }
-                    if app.bound_port.is_none() {
-                        // Currently, apps need to bind to a port before they can send from said port
-                        return ReturnCode::ERESERVE;
-                    }
-                    let next_tx = app.app_cfg.as_ref().and_then(|cfg| {
-                        if cfg.len() != 2 * mem::size_of::<UDPEndpoint>() {
-                            return None;
+                let res = self
+                    .apps
+                    .enter(appid, |app, _| {
+                        if app.pending_tx.is_some() {
+                            // Cannot support more than one pending tx per process.
+                            return Err(ErrorCode::BUSY);
+                        }
+                        if app.bound_port.is_none() {
+                            // Currently, apps need to bind to a port before they can send from said port
+                            return Err(ErrorCode::RESERVE);
                         }
+                        let next_tx = app.app_cfg.map_or(None, |cfg| {
+                            if cfg.len() != 2 * mem::size_of::<UDPEndpoint>() {
```

Comment:
This seems like a common idiom (2 * mem::size_of::); can it be abstracted into a method?

Reply:
we use both `2*mem::size_of::` and `mem::size_of`... we could abstract it into a method like `const fn size_of()` on `UDPEndpoint`, but that feels not worth it to me.

I did add a `use core::mem::size_of` so that each `2*mem::size_of<T>()` is now just `2*size_of<T>()`